### ISSUE TITLE
FATAL error: failed to verify transaction in bootstrapping: missing utxo while processing sync message

### ISSUE THREAD

**Issue by cryptovestor21** [ava-labs/avalanchego#1250]
**Describe the bug**
Two archive nodes we run have failed with the same error and are now stuck in a restart loop. They fail on fatal error as per the issue subject and the avalogs.txt linked below.

**To Reproduce**
- avalanche/1.9.9 [database=v1.4.5, rpcchainvm=23, commit=d755f872a4bf0de12297b3994b729ea684f78519]
- Install using the installation script at https://docs.avax.network/nodes/build/set-up-node-with-installer
- Attempt sync to chainhead
- Configure C-chain as below (might not be relevant, but it's my only customization from the install script)
```
{
  "snowman-api-enabled": false,
  "eth-apis": [
    "eth",
    "eth-filter",
    "net",
    "web3",
    "internal-eth",
    "internal-blockchain",
    "internal-transaction",
    "internal-tx-pool",
    "internal-account"
  ],
  "metrics-enabled": true,
  "pruning-enabled": false,
  "state-sync-enabled": false
}
```
**Expected behavior**
Expectation would be that the node continues to sync healthily rather than stalling on an error

**Logs**
[avaLogs.txt](https://github.com/ava-labs/avalanchego/files/11044956/avaLogs.txt)
[C.log](https://github.com/ava-labs/avalanchego/files/11044973/C.log)
[main.log](https://github.com/ava-labs/avalanchego/files/11044974/main.log)
[P.log](https://github.com/ava-labs/avalanchego/files/11044975/P.log)
[X.log](https://github.com/ava-labs/avalanchego/files/11044976/X.log)

**Operating System**
Ubuntu 22.04

**Additional context**
Some others have described similar issues on Discord https://discord.com/channels/578992315641626624/757576823570825316/1088211282496340018

**Comment by icculp** [comment]
To add some extra thought, issue seems to happen across versions. Had two nodes on 1.9.11-0.4.12, one got stuck after upgrading from 1.9.11 to 1.9.12, the other just after a simple restart on 1.9.11. Both nodes no longer functional, trying to resync from statesync on one of them and it appears the time just keeps increasing and isn't moving forward

**Comment by cryptovestor21** [comment]
A colleague has noted that they are still operating on v1.9.8 on their archive node.

**Comment by moraesjeremias** [comment]
@cryptovestor21 can you confirm if they are constantly running their AVAX nodes? Or if they were recently restarted?

**Comment by Liamlu28** [comment]
We are a node service provider running version 1.9.10, tried to upgrade to 1.9.11 and 1.9.12, but it doesn't work. Trying to downgrade the version 1.9.8 also does not work

**Comment by suntzu93** [comment]
Confirming that I also encountered a similar issue on 1.9.12 and 1.9.13. I upgraded from 1.9.8 and now even rolling back to 1.9.8 crashes with a similar error

```
[03-23|00:51:15.385] INFO server/server.go:269 adding route {"url": "/ext/bc/2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM", "endpoint": "/events"}
[03-23|00:51:15.385] INFO server/server.go:269 adding route {"url": "/ext/bc/2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM", "endpoint": ""}
[03-23|00:51:15.386] INFO <X Chain> bootstrap/bootstrapper.go:348 starting bootstrap
[03-23|00:51:45.536] INFO <C Chain> common/bootstrapper.go:244 bootstrapping started syncing {"numVerticesInFrontier": 2}
[03-23|00:51:47.601] INFO <X Chain> common/bootstrapper.go:244 bootstrapping started syncing {"numVerticesInFrontier": 33}
[03-23|00:51:47.603] INFO <X Chain> bootstrap/bootstrapper.go:578 executing transactions
[03-23|00:51:47.613] ERROR <X Chain> bootstrap/tx_job.go:101 transaction failed verification during bootstrapping {"txID": "2jB8AZPTEvrEE8wsB2BMSwgX1vwFhxsnSfQxj3QAfLHA4zLLsB", "error": "not found"}
[03-23|00:51:47.613] FATAL <X Chain> handler/handler.go:330 shutting down chain {"reason": "received an unexpected error", "error": "failed to execute job 2jB8AZPTEvrEE8wsB2BMSwgX1vwFhxsnSfQxj3QAfLHA4zLLsB due to failed to verify transaction in bootstrapping: not found while processing sync message: {%!s(*message.inboundMessage=&{[116 39 4 250 121 27 78 121 190 58 116 110 234 95 57 233 235 56 148 56] 15 0xc03647d830 {0 9223372036854775807 0x34230a0} <nil> 0}) ENGINE_TYPE_AVALANCHE}"}
```

**Comment by jsvisa** [comment]
We meet the same problem after updating to v1.10.12:


```
Mar 23 01:02:16 avalanchego[26988]: [03-23|01:02:16.136] ERROR <X Chain> bootstrap/tx_job.go:101 transaction failed verif[32/1846$uring bootstrapping {"txID": "2jB8AZPTEvrEE8wsB2BMSwgX1vwFhxsnSfQxj3QAfLHA4zLLsB", "error": "not found"}
Mar 23 01:02:16 avalanchego[26988]: [03-23|01:02:16.136] FATAL <X Chain> handler/handler.go:330 shutting down chain {"reason": "received an unexpected error", "error": "failed to execute job 2jB8AZPTEvrEE8wsB2BMSwgX1vwFhxsnSfQxj3QAfLHA4zLLsB due to failed to verify transaction in bootstrapping: not found while processing sync message: {%!s(*message.inboundMessage=&{[46 201 110 83 51 249 185 59 57 20 149 216 247 14 0 8 191 220 183 221] 15 0xc016368b40 {0 9223372036854775807 0x34360c0} <nil> 0}) ENGINE_TYPE_AVALANCHE}"}
```

**Comment by kim201212** [comment]
https://github.com/ava-labs/avalanchego/releases/tag/v1.9.14
Upgrading seems to fix the problem.

**Comment by jsvisa** [comment]
thanks, and upgrade to v.19.14 works fine

> https://github.com/ava-labs/avalanchego/releases/tag/v1.9.14 Upgrading seems to fix the problem.

**Comment by cryptovestor21** [comment]
Resolved on v1.9.14, closing this issue out.
